docs/gc.md

```diff
@@ -133,9 +132,9 @@ A global sweep operation also runs to reclaim memory so that it can create empty
 
 With the `balanced` policy, a global GC cycle is sometimes required in addition to the global mark operations and partial GC cycle. This global GC cycle is rare, occurring only in very tight memory conditions when other GC cycles cannot free enough memory on the heap.
 
-Most objects are easily contained within the minimum region size of 512 KB. However, to support large arrays, which cannot be contained in a region, the `balanced` GC policy uses an *arraylet* representation in the heap. For more information about structure and layout, see [Arraylets](allocation.md#arraylets).
+Before the 0.53.0 release, to support large arrays, which cannot be contained in a region, the `balanced` GC policy used an *arraylet* representation in the heap. With arraylets, JNI access to array data might involve reconstituting arraylets as contiguous arrays, which can significantly slow down processing. For more information about structure and layout, see [Arraylets](allocation.md#arraylets).
 
-:fontawesome-solid-pencil:{: .note aria-hidden="true"} **Note:** With arraylets, JNI access to array data might involve reconstituting arraylets as contiguous arrays, which can significantly slow down processing.
+All arrays are contiguous, but now if larger than a region size, the arrays are stored into a separate area. That area, the *Offheap*, is a few times larger than the main heap to prevent fragmentation, but it is only committed for the live arrays currently stored there. While increase of virtual address space will be observed, the total amount of committed memory (resident) will typically stay the same or increase marginally.
```

Comment:
I don't want to go too deep into details, but slightly more correct is to say that all array's data is contiguous. And than we can say that array's data  (rather than whole array) is stored into a separate area.

To clarify: array's header is always in the main heap, whether the data is adjacent to the header (small arrays, data in main heap) or non-adjacent to the header (larger array, data in Offheap).  We will have a blog providing more details - no need for that level of details here, but perhaps we could use array's data - not sure myself?